scripts/cbmc_json_parser.py

```diff
@@ -0,0 +1,398 @@
+        print_to_terminal(output_message)
+
+        # Delete temp Json file after displaying result
+        clear_json_file(json_file)
+    else:
+        # DynTrait tests generate a non json output due to "Invariant check failed" error
+        # For these cases, we just produce the cbmc output unparsed
+        # TODO: Parse these non json outputs from CBMC
+        non_json_cbmc_output_handler(log_file)
+    return
+
+# Write the json file to disk for easy parsing
+def save_json_file(log_file):
+
+    # TODO : Change to use the source file name as the temp file name too
+    # Create a temp json file for every source file as testing can be parallelized
+    temp_file_name = str(random.randint(10,1000000))
+    saved_json = "/home/ubuntu/rmc/scripts/temp_json/" + temp_file_name + "_temp.json"
+    try:
+        with open(saved_json, "w", encoding='utf-8') as outfile:
+            json.dump(log_file, outfile)
+    except FileNotFoundError as e:
+        print("Error in writing JSON file", e)
+    return saved_json
+
+# Check if the blob is in json format for parsing
+def is_json(myjson):
+    try:
+        json.loads(myjson)
+    except ValueError:
+        return False
+    return True
+
+# Function for printing temporary json objects for debugging purposes
+def print_json_file(input_json_file):
+    with open(input_json_file) as infile:
+        data = json.load(infile)
+        print(data)
+    return
+
+# Function to reformat CBMC blob into a parseable format
+def restructure_json_file(log_file):
+    """
+    CBMC's response blob is difficult to parse by itself, so it needs to be re-fit into
+    another temporary dictionary structure and 
+
+    Parameters -
+        log_file - log_file is the initial cbmc blob object which even in JSON format, 
+                    is difficult to parse and in some cases, is not in JSON format.
+
+    Returns - 
+        temp_dict - The restructured dictionary which is easy to parse
+
+    """
+    
+    # The restructured dictionary which is easy to parse
+    temp_dict = {}
+    try:
+        with open(log_file) as infile:
+            data = json.load(infile)
+            temp_dict["response"] = data
+    except FileNotFoundError:
+        print("CBMC File not found")
+        return {}
+    except JSONDecodeError:
+        # print out specific information about the extra data (like the lines before it)
+        print("JSON File not being parsed", )
+        return {}
+
+    # TODO: Check if passing dictionaries around is optimal or if there are alternate ways to 
+    # restructure and parse cbmc blob.
+    return temp_dict
+
+# Get Information about the Solver from the Json file written
+def extract_solver_information(json_object):
+    
+    # Solver information is all the fields in the json object which are not related to the results
+    solver_information = []
+    try:
+        responses = json_object["response"]
+    except:
+        print("No such Value called \"response\"")
```

Comment:
There is no point of keeping executing this function if there was an exception here, right?